flink-streaming-java/src/main/java/org/apache/flink/streaming/api/functions/sink/filesystem/StreamingFileSink.java

```diff
@@ -314,25 +332,30 @@ private BulkFormatBuilder(
 			this.partFileSuffix = Preconditions.checkNotNull(partFileSuffix);
 		}
 
-		public StreamingFileSink.BulkFormatBuilder<IN, BucketID> withBucketCheckInterval(long interval) {
-			return new BulkFormatBuilder<>(basePath, writerFactory, bucketAssigner, interval, bucketFactory, partFilePrefix, partFileSuffix);
+		public T withBucketCheckInterval(long interval) {
+			this.bucketCheckInterval = interval;
+			return self();
 		}
 
-		public <ID> StreamingFileSink.BulkFormatBuilder<IN, ID> withBucketAssigner(BucketAssigner<IN, ID> assigner) {
-			return new BulkFormatBuilder<>(basePath, writerFactory, Preconditions.checkNotNull(assigner), bucketCheckInterval, new DefaultBucketFactoryImpl<>(), partFilePrefix, partFileSuffix);
+		public T withBucketAssigner(BucketAssigner<IN, BucketID> assigner) {
```

Comment:
Here you are missing the parameter type `ID`. 

As described also in the comments here https://issues.apache.org/jira/browse/FLINK-13843, this method for the bulk formats and the `withBucketAssignerAndPolicy ()` for the row formats, are the ones that allow the user to change the default type of the `BucketID`, so they should stay as they are.

Reply:
@yxu-valleytider  Yes, this is why I am mentioned it here. The same holds for the `rowFormat` builder above where you explicitly cast. I am not so sure if this approach would work and we may need to investigate this further.

Reply:
@kl0u Did a small experiment with a non-string BucketID assigner :    
```
class NonStringBucketIdAssigner implements BucketAssigner<Message, Integer> {
    ...
}
```
and 
```
CustomizedStreamingFileSink.forBulkFormat(...)
            .withBucketAssigner(new NonStringBucketIdAssigner())
            .build();
```

The casting appears to be successful as long as the `assigner` parameter can be casted to `BucketAssigner`.  It makes sense considering the effect of unbounded type erasure.  

Also updating the PR to reflect this in the casting statement.   If the casting is truly a concern we could keep the original method with different method names, e.g., `newBuilderWithBucketAssigner`.